lib/plugins/aws/customResources/resources/utils.js

```diff
@@ -90,11 +91,53 @@ function wait(ms) {
   return new Promise(resolve => setTimeout(resolve, ms));
 }
 
+const MAX_AWS_REQUEST_TRY = (() => {
+  const DEFAULT_MAX_AWS_REQUEST_TRY = 4;
+  const userValue = Number(process.env.SLS_AWS_REQUEST_MAX_RETRIES);
+  return userValue >= 0 ? userValue : DEFAULT_MAX_AWS_REQUEST_TRY;
+})();
+
+const getServiceInstance = nameInput => {
+  let name;
+  let params = {};
+  if (typeof nameInput === 'string') {
+    name = nameInput;
+  } else {
+    name = nameInput.name;
+    params = nameInput.params;
+  }
+  return new AWS[name](params);
+};
+
+async function awsRequest(service, method, ...args) {
+  const serviceInstance = getServiceInstance(service);
+  const callAws = requestTry => {
+    return serviceInstance[method](...args)
+      .promise()
+      .then(
+        result => result,
+        error => {
+          if (
+            requestTry < MAX_AWS_REQUEST_TRY &&
+            error &&
+            (error.statusCode === 429 ||
+              (error.retryable && error.statusCode !== 403 && error.code !== 'CredentialsError'))
```

Comment:
I think we can limit this to `(error.statusCode === 429 || error.retryable)`, as in Lambda we should not expect credential errors, so it's not the case to consider